fury/ui/elements.py

```diff
@@ -3052,3 +3055,231 @@ def directory_click_callback(self, i_ren, _obj, listboxitem):
+            line_actor._points.SetPoint(i, new_x, new_y, 0.0)
+
+        line_actor._polygonPolyData.SetPoints(line_actor._points)
+        mapper = PolyDataMapper2D()
+        mapper = set_input(mapper, line_actor._polygonPolyData)
+
+        line_actor.actor.SetMapper(mapper)
+
+    def resize(self, size):
+        """Resize the UI.
+        """
+        if self.shape_type == "line":
+            hyp = np.hypot(size[0], size[1]) * (-1 if size[0] < 0 else 1)
+            self.shape.resize((hyp, 2))
+            self.rotate_line(deg=np.arctan(size[1]/size[0]))
+
+        elif self.shape_type == "quad":
+            self.shape.resize(size)
+
+        elif self.shape_type == "circle":
+            hyp = np.hypot(size[0], size[1])
+            self.shape.outer_radius = hyp
+
+    def left_button_pressed(self, i_ren, _obj, shape):
+        click_pos = np.array(i_ren.event.position)
+        self._drag_offset = click_pos - self.position
+        i_ren.event.abort()
+
+    def left_button_dragged(self, i_ren, _obj, shape):
+        if self._drag_offset is not None:
+            click_position = np.array(i_ren.event.position)
+            new_position = click_position - self._drag_offset
+            self.position = new_position
+        i_ren.force_render()
+
+
+class DrawPanel(UI):
+    def __init__(self, size=(400, 400), position=(0, 0)):
+        """Init this UI element.
+
+        Parameters
+        ----------
+        size : (int, int)
+            Width and height in pixels of this UI component.
+        position : (float, float)
+            (x, y) in pixels.
+        """
+        self.panel_size = size
+        super(DrawPanel, self).__init__(position)
+        self.shape_list = []
+        self.current_mode = "selection"
+
+    def _setup(self):
+        """Setup this UI component.
+
+        Create a Canvas(Panel2D).
+        """
+        self.canvas = Panel2D(size=self.panel_size)
+        self.canvas.background.on_left_mouse_button_pressed = self.left_button_pressed
+        self.canvas.background.on_left_mouse_button_dragged = self.left_button_dragged
+
+        mode_data = {
+            "line": "pencil.png",
+            "quad": "stop2.png",
+            "circle": "circle-up.png"
+        }
+
+        mode_panel_size = (len(mode_data) * 40, 40)
+        self.mode_panel = Panel2D(size=mode_panel_size, color=(0.5, 0.5, 0.5))
+        btn_pos = np.array([0, 0])
+        padding = 5
+
+        for mode in mode_data:
+            btn = Button2D(icon_fnames=[(mode, read_viz_icons(fname=mode_data[mode]))])
+
+            def mode_selector(i_ren, _obj, btn):
+                self.current_mode = btn.icon_names[0]
+
+            btn.on_left_mouse_button_pressed = mode_selector
+
+            self.mode_panel.add_element(btn, btn_pos+padding)
+            btn_pos[0] += btn.size[0]+padding
+
+        self.canvas.add_element(self.mode_panel, (0, 0))
+
+        self.mode_text = TextBlock2D(text="Mode: selection")
```

Comment:
Here in the setup, the `self.current_mode` isn't accessible as we have to change the text of `TextBlock2D` whenever the mode is changed, which is done after the creation of `TextBlock2D`.
I think it would be great to keep "Mode = " (blank) at the start because there isn't any significance of selection mode.

Reply:
Added "Select appropriate drawing mode using below icon" message on Startup.